hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/time/TimeFormatter.java

```diff
@@ -0,0 +1,54 @@
+package pl.allegro.tech.hermes.management.infrastructure.time;
+
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+
+@Component
+public class TimeFormatter {
+
+    private static final Logger logger = LoggerFactory.getLogger(TimeFormatter.class);
+
+    private final String dateTimeFormatPattern = "yyyy-MM-dd HH:mm:ss";
+
+    private final PeriodFormatter hourFormatter;
+    private final DateFormat dateFormat;
+
+    public TimeFormatter() {
+        hourFormatter = new PeriodFormatterBuilder()
+            .appendPrefix("-")
+            .appendHours()
+            .appendSuffix("h")
+            .toFormatter();
+
+        dateFormat = new SimpleDateFormat(dateTimeFormatPattern);
+    }
+
+    public Long parse(String formattedTime) {
+        try {
+            Period period = hourFormatter.parsePeriod(formattedTime);
+            return LocalDateTime.now().minusHours(period.getHours()).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
```

Comment:
I have also checked it manually and this is ok thanks toInstant() which represents some point on the time line. Begin of the time line is 1970-01-01T00:00:00Z.